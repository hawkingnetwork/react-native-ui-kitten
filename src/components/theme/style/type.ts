export interface StyleType {
  [key: string]: any;
}

export enum Interaction {
  HOVER = 'hover',
  ACTIVE = 'active',
  FOCUSED = 'focused',
  INDETERMINATE = 'indeterminate',
  VISIBLE = 'visible',
}

export enum State {
  CHECKED = 'checked',
  SELECTED = 'selected',
  DISABLED = 'disabled',
}
