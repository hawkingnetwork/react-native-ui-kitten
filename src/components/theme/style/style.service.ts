/**
 * @license
 * Copyright Akveo. All Rights Reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 */

import { ThemedStyleType } from '@eva-design/dss';
import { StyleType } from './type';
import { getThemeValue } from '../theme/theme.service';
import { ThemeType } from '../theme/type';

export function createThemedStyle(mapping: ThemedStyleType, theme: ThemeType): StyleType {
  return Object.keys(mapping).reduce((acc: StyleType, current: string): StyleType => {
    const mappingValue: any = mapping[current];

    return { ...acc, [current]: getThemeValue(mappingValue, theme, mappingValue) };
  }, {});
}
