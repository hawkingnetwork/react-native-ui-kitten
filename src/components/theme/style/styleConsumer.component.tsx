/**
 * @license
 * Copyright Akveo. All Rights Reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 */

import React from 'react';
import hoistNonReactStatics from 'hoist-non-react-statics';
import { ThemeStyleType } from '@eva-design/dss';
import { StyleConsumerService } from './styleConsumer.service';
import {
  Interaction,
  StyleType,
} from './type';
import { MappingContext } from '../mapping/mappingContext';
import { ThemeContext } from '../theme/themeContext';
import { ThemeType } from '../theme/type';

interface PrivateProps<T> {
  forwardedRef?: React.Ref<T>;
}

export interface StyledComponentProps {
  appearance?: string;
  theme?: ThemeType;
  themedStyle?: StyleType;
  dispatch?: (interaction: Interaction[]) => void;
}

interface State {
  interaction: Interaction[];
}

export interface ContextProps {
  style: ThemeStyleType;
  theme: ThemeType;
}

export type StyledComponentClass<P> = React.ComponentClass<StyledComponentProps & P>;

/**
 * `styled` is a High Order Function which is used to apply style mapping on component.
 *
 * Requires component to have `styledComponentName` string property which defines
 * corresponding component name in mapping. (e.g 'Button' for Button component).
 * Returns component class which can be used as styled component.
 *
 * @property {string} appearance - Determines style appearance of component. Default is provided by mapping.
 *
 * @property {ThemeType} theme - Determines theme used to style component.
 *
 * @property {StyleType} themedStyle - Determines component style for it's current state.
 *
 * @property {(interaction: Interaction[]) => void} dispatch - Determines function
 * for dispatching current state of component. This is designed to be used as style request function.
 * Calls component re-render if style for requested state differ from current.
 *
 * @param Component - Type: {ComponentType}. Determines class or functional component to be styled.
 *
 * @overview-example Declaring Styled Component
 *
 * ```
 * import React from 'react';
 * import { TouchableOpacity } from 'react-native';
 * import { styled, Interaction } from '@ui-kitten/components';
 *
 * class Button extends React.Component {
 *
 *   // Define component name used in `mapping`
 *   static styledComponentName = 'Button';
 *
 *   onPressIn = (e) => {
 *     // Request styles for `active` state and re-render
 *
 *     this.props.dispatch([Interaction.ACTIVE]);
 *
 *     if(this.props.onPressIn) {
 *       this.props.onPressIn(e);
 *     }
 *   };
 *
 *   onPressOut = (e) => {
 *     // Request styles for default state and re-render
 *
 *     this.props.dispatch([]);
 *
 *     if(this.props.onPressOut) {
 *       this.props.onPressOut(e);
 *     }
 *   };
 *
 *   render() {
 *     // Retrieve styles for current state from props (provided with themedStyle prop)
 *     // And apply it with saving priority of `style` prop
 *
 *     const { style, themedStyle, ...restProps } = this.props;
 *
 *     return (
 *       <TouchableOpacity
 *         {...restProps}
 *         style={[themedStyle, style]}
 *         onPressIn={this.onPressIn}
 *         onPressOut={this.onPressOut}
 *       />
 *     );
 *   }
 * }
 *
 * export const StyledButton = styled(Button);
 * ```
 *
 * @overview-example Styled Component Usage
 *
 * ```
 * import React from 'react';
 * import { StyledButton } from './path-to/styledButton.component';
 *
 * export const StyledButtonShowcase = (props) => (
 *   <StyledButton {...props}/>
 * );
 * ```
 */
export const styled = <P extends object>(Component: React.ComponentType<P>): StyledComponentClass<P> => {

  // @ts-ignore
  if (!Component.styledComponentName) {
    console.warn('Styled components should specify corresponding style name.');
    return null;
  }

  type WrappingProps = PrivateProps<WrappedElementInstance> & WrappedProps;
  type WrappedProps = StyledComponentProps & P;
  type WrappingElement = React.ReactElement<WrappingProps>;
  type WrappedElement = React.ReactElement<WrappedProps>;
  type WrappedElementInstance = React.ReactInstance;

  class Wrapper extends React.Component<WrappingProps, State> {

    public state: State = {
      interaction: [],
    };

    private init: boolean = false;

    // Yes. This is not static because it is calculated once we got some meta from style context.
    private defaultProps: StyledComponentProps;
    private service: StyleConsumerService;

    private onInit = (context: ContextProps) => {

      // @ts-ignore
      this.service = new StyleConsumerService(Component.styledComponentName, context);
      this.defaultProps = this.service.createDefaultProps();

      this.init = true;
    };

    private onDispatch = (interaction: Interaction[]) => {
      this.setState({ interaction });
    };

    private withStyledProps = (source: P, context: ContextProps): WrappedProps => {
      const { interaction } = this.state;

      const props: WrappingProps = { ...this.defaultProps, ...source };

      return this.service.withStyledProps(props, context, interaction);
    };

    private renderWrappedElement = (context: ContextProps): WrappedElement => {
      if (!this.init) {
        this.onInit(context);
      }

      const { forwardedRef, ...restProps } = this.props;
      const props: P & StyledComponentProps = this.withStyledProps(restProps as P, context);

      return (
        <Component
          {...props}
          ref={forwardedRef}
          dispatch={this.onDispatch}
        />
      );
    };

    public render(): React.ReactNode {
      const StyledElement = this.renderWrappedElement;

      return (
        <MappingContext.Consumer>{(styles: ThemeStyleType): WrappedElement => (
          <ThemeContext.Consumer>{(theme: ThemeType): WrappedElement => (
            <StyledElement style={styles} theme={theme}/>
          )}</ThemeContext.Consumer>
        )}</MappingContext.Consumer>
      );
    }
  }

  const WrappingElement = (props: WrappingProps, ref: React.Ref<WrappedElementInstance>): WrappingElement => {
    return (
      // @ts-ignore

      <Wrapper
        {...props}
        forwardedRef={ref}
      />
    );
  };

  const ResultComponent = React.forwardRef<WrappedElementInstance, WrappingProps>(WrappingElement);

  ResultComponent.displayName = Component.displayName || Component.name;

  // @ts-ignore
  hoistNonReactStatics(ResultComponent, Component);

  // @ts-ignore
  return ResultComponent;
};
