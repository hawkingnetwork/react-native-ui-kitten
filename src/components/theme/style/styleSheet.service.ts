import { StyleSheet as RNStyleSheet } from 'react-native';

type ThemeValue = string;

export type ThemeType = Record<string, ThemeValue>;
export type StyleType = Record<string, any>;
export type Styles<T> = RNStyleSheet.NamedStyles<T>;

export class StyleSheet {

  static createThemedStyles = <T>(styles: Styles<T>, theme: ThemeType): T => {
    return Object.keys(styles).reduce((acc: T, key: string): T => {
      return { ...acc, [key]: StyleSheet.createThemedStyle(styles[key], theme)  };
    }, {} as T);
  };

  static createThemedStyle = (style: StyleType, theme: ThemeType): StyleType => {
    return Object.keys(style).reduce((acc: StyleType, key: string): StyleType => {
      const value: any = style[key];
      return { ...acc, [key]: StyleSheet.getThemeValue(value, theme, value) };
    }, {});
  };

  static createCompiledTheme = (theme: ThemeType): ThemeType => {
    return Object.keys(theme).reduce((acc: ThemeType, key: string): ThemeType => {
      return { ...acc, [key]: StyleSheet.getThemeValue(key, theme, key) };
    }, {});
  };

  static getThemeValue = (name: string, theme: ThemeType, fallback?: ThemeValue): ThemeValue | undefined => {
    if (StyleSheet.isReference(name)) {
      const themeKey: string = StyleSheet.toThemeKey(name);
      return StyleSheet.findThemeValue(themeKey, theme) || fallback;
    }

    return StyleSheet.findThemeValue(name, theme) || fallback;
  };

  static findThemeValue = (name: string, theme: ThemeType): ThemeValue | undefined => {
    const value: ThemeValue = theme[name];

    if (StyleSheet.isReference(value)) {
      const themeKey: string = StyleSheet.toThemeKey(value);
      return StyleSheet.findThemeValue(themeKey, theme);
    }

    return value;
  };

  /**
   * @returns true if theme value references to another
   */
  static isReference = (value: ThemeValue): boolean => {
    return `${value}`.startsWith('$');
  };

  /**
   * Transforms reference key to theme key
   */
  static toThemeKey = (value: ThemeValue): string => {
    return `${value}`.substring(1);
  };
}
