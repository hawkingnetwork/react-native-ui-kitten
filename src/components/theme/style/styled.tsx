/**
 * @license
 * Copyright Akveo. All Rights Reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 */

import React from 'react';
import hoistNonReactStatics from 'hoist-non-react-statics';
import { ThemeStyleType } from '@eva-design/dss';
import { StyleConsumerService } from './styleConsumer.service';
import {
  Interaction,
  StyleType,
} from './style.service';
import { MappingContext } from '../mapping/mappingContext';
import { ThemeContext } from '../theme/themeContext';
import { ThemeType } from '../theme/theme.service';

interface PrivateProps<T> {
  forwardedRef?: React.Ref<T>;
}

export interface StyledComponentProps {
  appearance?: string;
  theme?: ThemeType;
  themedStyle?: StyleType;
  dispatch?: (interaction: Interaction[]) => void;
}

interface State {
  interaction: Interaction[];
}

export type StyledComponentClass<P> = React.ComponentClass<StyledComponentProps & P>;

/**
 * `styled` is a High Order Function which is used to apply style mapping on component.
 *
 * Requires component to have `styledComponentName` string property which defines
 * corresponding component name in mapping. (e.g 'Button' for Button component).
 * Returns component class which can be used as styled component.
 *
 * @property {string} appearance - Determines style appearance of component. Default is provided by mapping.
 *
 * @property {ThemeType} theme - Determines theme used to style component.
 *
 * @property {StyleType} themedStyle - Determines component style for it's current state.
 *
 * @property {(interaction: Interaction[]) => void} dispatch - Determines function
 * for dispatching current state of component. This is designed to be used as style request function.
 * Calls component re-render if style for requested state differ from current.
 *
 * @param Component - Type: {ComponentType}. Determines class or functional component to be styled.
 *
 * @overview-example StyledComponentSimpleUsage
 *
 * @overview-example StyledComponentStates
 *
 * @overview-example StyledComponentVariants
 */
export const styled = <P extends object>(Component: React.ComponentType<P>): StyledComponentClass<P> => {

  // @ts-ignore
  if (!Component.styledComponentName) {
    console.warn('Styled components should specify corresponding style name.');
    return null;
  }

  type WrappingProps = PrivateProps<WrappedElementInstance> & WrappedProps;
  type WrappedProps = StyledComponentProps & P;
  type WrappingElement = React.ReactElement<WrappingProps>;
  type WrappedElement = React.ReactElement<WrappedProps>;
  type WrappedElementInstance = React.ReactInstance;

  class Wrapper extends React.Component<WrappingProps, State> {

    public state: State = {
      interaction: [],
    };

    private init: boolean = false;

    // Yes. This is not static because it is calculated once we got some meta from style context.
    private defaultProps: StyledComponentProps;
    private service: StyleConsumerService;

    private onInit = (style: ThemeStyleType, theme: ThemeType): void => {
      // @ts-ignore
      this.service = new StyleConsumerService(Component.styledComponentName, style, theme);
      this.defaultProps = this.service.createDefaultProps();

      this.init = true;
    };

    private onDispatch = (interaction: Interaction[]): void => {
      this.setState({ interaction });
    };

    private withStyledProps = (source: P,
                               style: ThemeStyleType,
                               theme: ThemeType): WrappedProps => {
      const { interaction } = this.state;

      const props: WrappingProps = { ...this.defaultProps, ...source };

      return this.service.withStyledProps(props, style, theme, interaction);
    };

    private renderWrappedElement = (style: ThemeStyleType, theme: ThemeType): WrappedElement => {
      if (!this.init) {
        this.onInit(style, theme);
      }

      const { forwardedRef, ...restProps } = this.props;
      const props: P & StyledComponentProps = this.withStyledProps(restProps as P, style, theme);

      return (
        <Component
          {...props}
          ref={forwardedRef}
          dispatch={this.onDispatch}
        />
      );
    };

    public render(): React.ReactNode {
      return (
        <MappingContext.Consumer>{(style: ThemeStyleType): WrappedElement => (
          <ThemeContext.Consumer>{(theme: ThemeType): WrappedElement => {
            return this.renderWrappedElement(style, theme);
          }}</ThemeContext.Consumer>
        )}</MappingContext.Consumer>
      );
    }
  }

  const WrappingElement = (props: WrappingProps,
                           ref: React.Ref<WrappedElementInstance>): WrappingElement => {
    return (
      // @ts-ignore

      <Wrapper
        {...props}
        forwardedRef={ref}
      />
    );
  };

  const ResultComponent = React.forwardRef<WrappedElementInstance, WrappingProps>(WrappingElement);

  ResultComponent.displayName = Component.displayName || Component.name;

  // @ts-ignore
  hoistNonReactStatics(ResultComponent, Component);

  // @ts-ignore
  return ResultComponent;
};
