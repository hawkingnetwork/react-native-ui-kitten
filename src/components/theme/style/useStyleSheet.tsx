import {
  Styles,
  StyleSheet,
  ThemeType,
} from '../style/styleSheet.service';
import { useTheme } from '../theme/useTheme';

/**
 * Takes a theme provided by ApplicationProvider or ThemeProvider and applies it to style.
 *
 * @overview-example UseStyleSheetSimpleUsage
 */
export const useStyleSheet = <T extends Styles<T>>(styles: Styles<T>) => {
  return {
    create: (): T => {
      const theme: ThemeType = useTheme();
      return StyleSheet.createThemedStyles(styles, theme);
    },
  };
};
