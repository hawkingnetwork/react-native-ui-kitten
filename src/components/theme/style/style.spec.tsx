import React from 'react';
import {
  TouchableOpacity,
  View,
  ViewProps,
} from 'react-native';
import {
  fireEvent,
  render,
  RenderAPI,
  waitForElement,
} from 'react-native-testing-library';
import { ReactTestInstance } from 'react-test-renderer';
import {
  StyleProvider,
  StyleProviderProps,
} from './styleProvider.component';
import {
  styled,
  StyledComponentProps,
  ContextProps,
} from './styleConsumer.component';
import { StyleConsumerService } from './styleConsumer.service';
import { createThemedStyle } from './style.service';
import {
  Interaction,
  StyleType,
} from './type';
import {
  ThemedStyleType,
  ThemeType,
} from '../theme/type';
import {
  styles,
  theme,
  themeInverse,
} from '../support/tests';

const styleConsumerTestId = '@style/consumer';
const styleTouchableTestId = '@style/touchable';

interface ThemeChangingProviderProps extends StyleProviderProps {
  themeInverse: ThemeType;
}

class ThemeChangingProvider extends React.Component<ThemeChangingProviderProps> {

  public state = {
    styles: this.props.styles,
    theme: this.props.theme,
  };

  private onTouchablePress = () => {
    this.setState({
      theme: this.props.themeInverse,
    });
  };

  public render(): React.ReactElement<ThemeChangingProviderProps> {
    return (
      <StyleProvider {...this.state}>
        <TouchableOpacity
          testID={styleTouchableTestId}
          onPress={this.onTouchablePress}>
          {this.props.children}
        </TouchableOpacity>
      </StyleProvider>
    );
  }
}

interface TestComponentProps extends StyledComponentProps, ViewProps {
  disabled?: boolean;
}

class Test extends React.Component<TestComponentProps> {
  static styledComponentName: string = 'Radio';

  static defaultProps: Partial<TestComponentProps> = {
    testID: styleConsumerTestId,
  };

  public render(): React.ReactElement<ViewProps> {
    return (
      <View testID={this.props.testID}/>
    );
  }
}

class NonStyledComponent extends React.Component<TestComponentProps> {
  public render(): React.ReactElement<ViewProps> {
    return undefined;
  }
}


const json = (value: any): string => JSON.stringify(value);

describe('@style: consumer service methods check', () => {

  const context: ContextProps = {
    style: styles,
    theme: theme,
  };

  const service: StyleConsumerService = new StyleConsumerService('Radio', context);

  describe('* style mapping', () => {

    const derivedProps: StyledComponentProps & any = {
      appearance: 'default',
      disabled: true,
    };

    it('creates valid default props', () => {
      const value: StyledComponentProps = service.createDefaultProps();

      expect(json(value)).toEqual(json({
        appearance: 'default',
        size: 'medium',
      }));
    });

    it('creates valid themedStyle prop', () => {
      const defaultProps: StyledComponentProps = service.createDefaultProps();

      const props: StyledComponentProps = {
        ...defaultProps,
        ...derivedProps,
      };

      const value: StyledComponentProps = service.withStyledProps(props, context, [Interaction.ACTIVE]);

      expect(value.themedStyle).toMatchSnapshot();
    });

  });

});

describe('@style: service methods checks', () => {

  describe('* styling', () => {

    const mapping: ThemedStyleType = {
      prop1: 'blue-primary',
      prop2: 'blue-dark',
      prop3: 'gray-primary',
      prop4: 42,
    };

    it('* default theme', () => {
      const value: StyleType = createThemedStyle(mapping, theme);

      expect(value).toMatchSnapshot();
    });

  });

});

describe('@style: ui component checks', () => {

  it('* returns null if has no static `styledComponentName` property', () => {
    const styledComponent = styled(NonStyledComponent);

    expect(styledComponent).toBeNull();
  });

  it('* static methods are copied over', async () => {
    // @ts-ignore: test-case
    Test.staticMethod = function () {
    };
    const StyleConsumer = styled<TestComponentProps>(Test);

    // @ts-ignore: test-case
    expect(StyleConsumer.staticMethod).not.toBeUndefined();
  });

  it('* receives custom props', async () => {
    const StyleConsumer = styled<TestComponentProps>(Test);

    const component: RenderAPI = render(
      <StyleProvider styles={styles} theme={theme}>
        <StyleConsumer/>
      </StyleProvider>,
    );

    const styledComponent = component.getByTestId(styleConsumerTestId);

    expect(styledComponent.props.appearance).not.toBeNull();
    expect(styledComponent.props.appearance).not.toBeUndefined();
    expect(styledComponent.props.theme).not.toBeNull();
    expect(styledComponent.props.theme).not.toBeUndefined();
    expect(styledComponent.props.themedStyle).not.toBeNull();
    expect(styledComponent.props.themedStyle).not.toBeUndefined();
    expect(styledComponent.props.dispatch).not.toBeNull();
    expect(styledComponent.props.dispatch).not.toBeUndefined();
  });

  it('* default appearance styled properly', async () => {
    const StyleConsumer = styled<TestComponentProps>(Test);

    const component: RenderAPI = render(
      <StyleProvider styles={styles} theme={theme}>
        <StyleConsumer/>
      </StyleProvider>,
    );

    const withStateProp: RenderAPI = render(
      <StyleProvider styles={styles} theme={theme}>
        <StyleConsumer disabled={true}/>
      </StyleProvider>,
    );

    const styledComponent: ReactTestInstance = component.getByTestId(styleConsumerTestId);
    const withAppearanceComponent: ReactTestInstance = withStateProp.getByTestId(styleConsumerTestId);

    expect(styledComponent.props.themedStyle).toMatchSnapshot();
    expect(withAppearanceComponent.props.themedStyle).toMatchSnapshot();
  });

  it('* dispatch action works properly', async () => {
    const StyleConsumer = styled<TestComponentProps>(Test);

    const component: RenderAPI = render(
      <StyleProvider styles={styles} theme={theme}>
        <StyleConsumer/>
      </StyleProvider>,
    );

    const styledComponent = component.getByTestId(styleConsumerTestId);
    styledComponent.props.dispatch([Interaction.ACTIVE]);

    const styledComponentChanged = await waitForElement(() => {
      return component.getByTestId(styleConsumerTestId);
    });

    expect(styledComponentChanged.props.themedStyle).toMatchSnapshot();
  });

  it('* provides correct styles on theme change', async () => {
    const StyleConsumer = styled<TestComponentProps>(Test);

    const component: RenderAPI = render(
      <ThemeChangingProvider
        styles={styles}
        theme={theme}
        themeInverse={themeInverse}>
        <StyleConsumer/>
      </ThemeChangingProvider>,
    );

    const styledComponent: ReactTestInstance = component.getByTestId(styleConsumerTestId);

    expect(styledComponent.props.themedStyle).toMatchSnapshot();

    const touchableComponent: ReactTestInstance = component.getByTestId(styleTouchableTestId);

    fireEvent.press(touchableComponent);

    const styledComponentChanged: ReactTestInstance = await waitForElement(() => {
      return component.getByTestId(styleConsumerTestId);
    });

    expect(styledComponentChanged.props.themedStyle).toMatchSnapshot();
  });

});
