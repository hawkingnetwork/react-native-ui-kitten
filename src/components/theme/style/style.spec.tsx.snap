// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@style: consumer service methods check * style mapping creates valid themedStyle prop 1`] = `
Object {
  "borderColor": "#8992A3",
  "borderRadius": 18,
  "borderWidth": 2,
  "height": 36,
  "highlightBackgroundColor": "#DDE1EB",
  "highlightBorderRadius": 30,
  "highlightHeight": 60,
  "highlightWidth": 60,
  "selectBackgroundColor": "transparent",
  "selectBorderRadius": 12,
  "selectHeight": 24,
  "selectWidth": 24,
  "textColor": "#c8cedb",
  "textFontSize": 16,
  "textFontWeight": "500",
  "textMarginLeft": 12,
  "width": 36,
}
`;

exports[`@style: service methods checks * styling * default theme 1`] = `
Object {
  "prop1": "#3366FF",
  "prop2": "#2541CC",
  "prop3": "#A6AEBD",
  "prop4": 42,
}
`;

exports[`@style: ui component checks * default appearance styled properly 1`] = `
Object {
  "borderColor": "#A6AEBD",
  "borderRadius": 18,
  "borderWidth": 2,
  "height": 36,
  "highlightBackgroundColor": "transparent",
  "highlightBorderRadius": 30,
  "highlightHeight": 60,
  "highlightWidth": 60,
  "selectBackgroundColor": "transparent",
  "selectBorderRadius": 12,
  "selectHeight": 24,
  "selectWidth": 24,
  "textColor": "#000000",
  "textFontSize": 16,
  "textFontWeight": "500",
  "textMarginLeft": 12,
  "width": 36,
}
`;

exports[`@style: ui component checks * default appearance styled properly 2`] = `
Object {
  "borderColor": "#DDE1EB",
  "borderRadius": 18,
  "borderWidth": 2,
  "height": 36,
  "highlightBackgroundColor": "transparent",
  "highlightBorderRadius": 30,
  "highlightHeight": 60,
  "highlightWidth": 60,
  "selectBackgroundColor": "transparent",
  "selectBorderRadius": 12,
  "selectHeight": 24,
  "selectWidth": 24,
  "textColor": "#c8cedb",
  "textFontSize": 16,
  "textFontWeight": "500",
  "textMarginLeft": 12,
  "width": 36,
}
`;

exports[`@style: ui component checks * dispatch action works properly 1`] = `
Object {
  "borderColor": "#8992A3",
  "borderRadius": 18,
  "borderWidth": 2,
  "height": 36,
  "highlightBackgroundColor": "#DDE1EB",
  "highlightBorderRadius": 30,
  "highlightHeight": 60,
  "highlightWidth": 60,
  "selectBackgroundColor": "transparent",
  "selectBorderRadius": 12,
  "selectHeight": 24,
  "selectWidth": 24,
  "textColor": "#000000",
  "textFontSize": 16,
  "textFontWeight": "500",
  "textMarginLeft": 12,
  "width": 36,
}
`;

exports[`@style: ui component checks * provides correct styles on theme change 1`] = `
Object {
  "borderColor": "#A6AEBD",
  "borderRadius": 18,
  "borderWidth": 2,
  "height": 36,
  "highlightBackgroundColor": "transparent",
  "highlightBorderRadius": 30,
  "highlightHeight": 60,
  "highlightWidth": 60,
  "selectBackgroundColor": "transparent",
  "selectBorderRadius": 12,
  "selectHeight": 24,
  "selectWidth": 24,
  "textColor": "#000000",
  "textFontSize": 16,
  "textFontWeight": "500",
  "textMarginLeft": 12,
  "width": 36,
}
`;

exports[`@style: ui component checks * provides correct styles on theme change 2`] = `
Object {
  "borderColor": "#A6AEBD",
  "borderRadius": 18,
  "borderWidth": 2,
  "height": 36,
  "highlightBackgroundColor": "transparent",
  "highlightBorderRadius": 30,
  "highlightHeight": 60,
  "highlightWidth": 60,
  "selectBackgroundColor": "transparent",
  "selectBorderRadius": 12,
  "selectHeight": 24,
  "selectWidth": 24,
  "textColor": "#FFFFFF",
  "textFontSize": 16,
  "textFontWeight": "500",
  "textMarginLeft": 12,
  "width": 36,
}
`;
